pkg/backends/azurekeyvault.go

```diff
@@ -0,0 +1,100 @@
+package backends
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/keyvault/keyvault"
+	"path"
+	"strings"
+	"time"
+)
+
+// AzureKeyVault is a struct for working with an Azure Key Vault backend
+type AzureKeyVault struct {
+	Client keyvault.BaseClient
+}
+
+// NewAzureKeyVaultBackend initializes a new Azure Key Vault backend
+func NewAzureKeyVaultBackend(client keyvault.BaseClient) *AzureKeyVault {
+	return &AzureKeyVault{
+		Client: client,
+	}
+}
+
+// Login does nothing as a "login" is handled on the instantiation of the Azure SDK
+func (a *AzureKeyVault) Login() error {
+	return nil
+}
+
+// GetSecrets gets secrets from Azure Key Vault and returns the formatted data
+func (a *AzureKeyVault) GetSecrets(kvpath string, version string, _ map[string]string) (map[string]interface{}, error) {
+	kvpath = fmt.Sprintf("https://%s.vault.azure.net", kvpath)
```

Comment:
I would like to maybe understand this logic a little more. I see 6 different `for-loops` and wonder if there is a way that some of that can be combined. I imagine the performance on this will not be ideal, especially if there are a lot of different inline paths.

Reply:
In an inline scenario, i imagine we can just get the secret by name and then output just that so we dont have to run through this logic for every inline secret

Reply:
@werne2j Agree, I somewhere had an idea that it could be nice to use some metadata that is in Azure Key Vault and use it in some way perhaps. But you are right, if not a performance hit, at least a longer fetching time and the first 3 for loops can easily be removed. :+1:  Let me get back with an update to this PR after the weekend if its OK.

Reply:
I have hopefully cleaned out the code enough now and also added documentation. Let me know what you think :)

Reply:
So for Azure KV, the path isn't actually a `/` delimited path because otherwise you wouldn't have a valid URL there. Would you mind adding a comment to note this, something like:

```
// GetSecrets does blah blah blah
// For Azure KV, the `kvpath` is actually blah, because blah blah
```

I'm guessing the SDK will handle provide a helpful error message for errors caused by giving something like `/path/to/secret` for `kvpath`

This might be good to mention in the docs too - your call on that though, I'm not too familiar with Azure KV so maybe what I'm saying is obvious to a user of that service

Reply:
Updated text and in docs.